Index: lib/pages/profile/change_password.dart
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- lib/pages/profile/change_password.dart	(revision 32165f821e11270fa0f05d484502ef1c65c8c310)
+++ lib/pages/profile/change_password.dart	(date 1597005768000)
@@ -96,20 +96,26 @@
               print("Auth Credential: ${authCredential.toString()}");
               user.reauthenticateWithCredential(authCredential).then((result) {
                 setState(() {
-                  error = "Success: $result";
+                  error = "Authentication Success!";
                 });
-                user.updatePassword(_newPassword).then((result) {
-                  setState(() {
-                    error = "Password updated.";
-                  });
-                }).catchError((result) {
-                  setState(() {
-                    error = "An error occurred while changing the password: " + result.message;
+                if (_newPassword2 == _newPassword) {
+                  user.updatePassword(_newPassword).then((result) {
+                    setState(() {
+                      error = "Password had been updated successfully.";
+                    });
+                  }).catchError((result) {
+                    setState(() {
+                      error = "An error occurred while changing the password: " + result.message;
+                    });
                   });
-                });
+                } else {
+                  setState(() {
+                    error = "An error occurred while changing the password: Your new passwords are mismatched.";
+                  });
+                }
               }).catchError((result) {
                 setState(() {
-                  error = "[Old Password] " + result.message;
+                  error = "Authentication Failure: Your old password is invalid.";
                 });
               });
             },
@@ -223,7 +229,9 @@
                     ),
                   ),
                   obscureText: true,
-                  validator: (val) => val.isEmpty ? 'Enter your new password.' : null,
+                  validator: (val) => val.length  < 6
+                      ? 'Please enter minimum 6 characters'
+                      : null,
                   onChanged: (val) {
                     _newPassword = val;
                   },
@@ -277,12 +285,11 @@
                     ),
                   ),
                   obscureText: true,
-                  validator: (val) => val.isEmpty ? 'Enter your new password.' : null,
+                  validator: (val) => val.length  < 6
+                      ? 'Please enter minimum 6 characters'
+                      : null,
                   onChanged: (val) {
                     _newPassword2 = val;
-                    if (_newPassword2 == _newPassword) {
-                      print("new password matched");
-                    } else {}
                   },
                 ),
               ),
Index: lib/pages/profile/settings.dart
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- lib/pages/profile/settings.dart	(revision 32165f821e11270fa0f05d484502ef1c65c8c310)
+++ lib/pages/profile/settings.dart	(date 1597005019000)
@@ -164,10 +164,22 @@
                 child: Column(
                   children: <Widget>[
                     SizedBox(height: 15),
-                    CircleAvatar(
-//                      backgroundImage: NetworkImage(storedProfilePic),
-                      backgroundColor: Color(0xffE8E7E7),
-                      radius: 121/2,
+                    StreamBuilder<DocumentSnapshot>(
+                      stream: Firestore.instance.collection('users').document(uid).snapshots(),
+                      builder: (context, AsyncSnapshot<DocumentSnapshot> snapshot) {
+                        String dp = '';
+                        try {
+                          dp = snapshot.data["profilePic"];
+                        } catch (Exception) {}
+                        if (snapshot.data == null) return CircularProgressIndicator();
+                        return CircleAvatar(
+                          backgroundImage: dp.isNotEmpty
+                              ? NetworkImage(dp)
+                              : CircularProgressIndicator(),
+                          backgroundColor: Color(0xffE8E7E7),
+                          radius: 121/2,
+                        );
+                      }
                     ),
                     SizedBox(height: 15),
 //                    FlatButton(
@@ -217,14 +229,24 @@
                                     fontWeight: FontWeight.w500
                                 ),
                               ),
-                              subtitle: Text(
-                                storedUsername,
-                                style: TextStyle(
-                                    fontFamily: 'Poppins',
-                                    fontSize: 16,
-                                    color: Colors.grey[600],
-                                    fontWeight: FontWeight.w400
-                                ),
+                              subtitle: StreamBuilder<DocumentSnapshot>(
+                                stream: Firestore.instance.collection('users').document(uid).snapshots(),
+                                builder: (context, AsyncSnapshot<DocumentSnapshot> snapshot) {
+                                  String username = '';
+                                  try {
+                                    username = snapshot.data["username"];
+                                  } catch (Exception) {}
+                                  if (snapshot.data == null) return CircularProgressIndicator();
+                                  return Text(
+                                    username,
+                                    style: TextStyle(
+                                        fontFamily: 'Poppins',
+                                        fontSize: 16,
+                                        color: Colors.grey[600],
+                                        fontWeight: FontWeight.w400
+                                    ),
+                                  );
+                                }
                               ),
                               leading: Icon(Icons.person_outline, color: Color(0xffFFC000),),
                               trailing: Icon(Icons.keyboard_arrow_right),
@@ -247,14 +269,24 @@
                                     fontWeight: FontWeight.w500
                                 ),
                               ),
-                              subtitle: Text(
-                                storedBio,
-                                style: TextStyle(
-                                    fontFamily: 'Poppins',
-                                    fontSize: 16,
-                                    color: Colors.grey[600],
-                                    fontWeight: FontWeight.w400
-                                ),
+                              subtitle: StreamBuilder<DocumentSnapshot>(
+                                stream: Firestore.instance.collection('users').document(uid).snapshots(),
+                                builder: (context, AsyncSnapshot<DocumentSnapshot> snapshot) {
+                                  String bio = '';
+                                  try {
+                                    bio = snapshot.data["bio"];
+                                  } catch (Exception) {}
+                                  if (snapshot.data == null) return CircularProgressIndicator();
+                                  return Text(
+                                    bio,
+                                    style: TextStyle(
+                                        fontFamily: 'Poppins',
+                                        fontSize: 16,
+                                        color: Colors.grey[600],
+                                        fontWeight: FontWeight.w400
+                                    ),
+                                  );
+                                }
                               ),
                               leading: Icon(Icons.chat_bubble_outline, color: Color(0xffFFC000),),
                               trailing: Icon(Icons.keyboard_arrow_right),
Index: lib/pages/profile/my_profile.dart
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- lib/pages/profile/my_profile.dart	(revision 32165f821e11270fa0f05d484502ef1c65c8c310)
+++ lib/pages/profile/my_profile.dart	(date 1597004958000)
@@ -91,10 +91,22 @@
                       SizedBox(width: 28.0),
                       Column(
                           children: <Widget> [
-                            CircleAvatar(
-//                              backgroundImage: NetworkImage(storedProfilePic),
-                              backgroundColor: Color(0xffE8E7E7),
-                              radius: 121/2,
+                            StreamBuilder<DocumentSnapshot>(
+                              stream: Firestore.instance.collection('users').document(uid).snapshots(),
+                              builder: (context, AsyncSnapshot<DocumentSnapshot> snapshot) {
+                                String dp = '';
+                                try {
+                                  dp = snapshot.data["profilePic"];
+                                } catch (Exception) {}
+                                if (snapshot.data == null) return CircularProgressIndicator();
+                                return CircleAvatar(
+                                  backgroundImage: dp.isNotEmpty
+                                      ? NetworkImage(dp)
+                                      : CircularProgressIndicator(),
+                                  backgroundColor: Color(0xffE8E7E7),
+                                  radius: 121/2,
+                                );
+                              }
                             ),
                             SizedBox(height: 5),
                             Container(
@@ -145,13 +157,22 @@
                         mainAxisAlignment: MainAxisAlignment.start,
                         crossAxisAlignment: CrossAxisAlignment.start,
                         children: <Widget>[
-                          Text(
-                            storedUsername,
-                            style: TextStyle(
-                              fontSize: 21.0,
-                              fontFamily: 'Poppins',
-                              fontWeight: FontWeight.w700,
-                            ),
+                          StreamBuilder<DocumentSnapshot>(
+                            stream: Firestore.instance.collection('users').document(uid).snapshots(),
+                            builder: (context, AsyncSnapshot<DocumentSnapshot> snapshot) {
+                              String username = '';
+                              try {
+                                username = snapshot.data["username"];
+                              } catch (Exception) {}
+                                if (snapshot.data == null) return CircularProgressIndicator();
+                                return new Text(username,
+                                  style: TextStyle(
+                                    fontSize: 21.0,
+                                    fontFamily: 'Poppins',
+                                    fontWeight: FontWeight.w700,
+                                  ),
+                                );
+                            }
                           ),
                           Text(
                             'Life Lover',
@@ -162,15 +183,25 @@
                               color: Color(0xff7A8FA6),
                             ),
                           ),
-                          Text(
-                            storedBio,
-                            style: TextStyle(
-                              fontSize: 16,
-                              fontFamily: 'Poppins',
-                              fontWeight: FontWeight.w400,
-                              color: Color(0xff7A8FA6),
-                            ),
-                            maxLines: null,
+                          StreamBuilder<DocumentSnapshot>(
+                            stream: Firestore.instance.collection('users').document(uid).snapshots(),
+                            builder: (context, AsyncSnapshot<DocumentSnapshot> snapshot) {
+                              String bio = '';
+                              try {
+                                bio = snapshot.data["bio"];
+                              } catch (Exception) {}
+                              if (snapshot.data == null) return CircularProgressIndicator();
+                              return Text(
+                                bio,
+                                style: TextStyle(
+                                  fontSize: 16,
+                                  fontFamily: 'Poppins',
+                                  fontWeight: FontWeight.w400,
+                                  color: Color(0xff7A8FA6),
+                                ),
+                                maxLines: null,
+                              );
+                            }
                           ),
                         ],
                       )
